src/chainparamsbase.cpp

```diff
@@ -94,22 +93,26 @@ CBaseChainParams::Network NetworkIdFromCommandLine()
     bool fTestNet = GetBoolArg("-testnet", false);
 
     if (fTestNet && fRegTest)
-        return CBaseChainParams::MAX_NETWORK_TYPES;
+        throw std::runtime_error("Invalid combination of -regtest and -testnet.\n");
     if (fRegTest)
         return CBaseChainParams::REGTEST;
     if (fTestNet)
         return CBaseChainParams::TESTNET;
-    return CBaseChainParams::MAIN;
+
+    std::string chain = GetArg("-chain", "main");
+    if (chain == "main")
+        return CBaseChainParams::MAIN;
```

Comment:
What if these would be defined as string constants, e.g.:
```
    const std::string CBaseChainParams::MAIN = "main";
    const std::string CBaseChainParams::TESTNET = "testnet";
    const std::string CBaseChainParams::REGTEST = "regtest";
```
... I guess we could get rid of the `CBaseChainParams::Network` enumeration type completely, making `SelectParams` take a string. This kind of switching code would be no longer necessary apart from the data structure selection in SelectParams itself.